packages/api/src/command/medical/patient/convert-fhir-bundle.ts

```diff
@@ -0,0 +1,203 @@
+  patient: Pick<Patient, "id" | "cxId" | "data">;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+  conversionType: ConsolidationConversionType;
+}): Promise<string> {
+  const lambdaName = Config.getFHIRToMedicalRecordLambdaName();
+
+  // create file name
+  const fileName = createFileName({
+    cxId: patient.cxId,
+    patientId: patient.id,
+    resources,
+    dateFrom,
+    dateTo,
+  });
+
+  // store on S3
+  const s3 = makeS3Client(Config.getAWSRegion());
+  const emptyMetaProp = "na";
+  await s3
+    .putObject({
+      Bucket: Config.getMedicalDocumentsBucketName(),
+      Key: fileName,
+      Body: JSON.stringify(bundle),
+      ContentType: "application/json",
+      Metadata: {
+        patientId: patient.id,
+        cxId: patient.cxId,
+        resources: resources?.toString() ?? emptyMetaProp,
+        dateFrom: dateFrom ?? emptyMetaProp,
+        dateTo: dateTo ?? emptyMetaProp,
+        conversionType,
+      },
+    })
+    .promise();
+
+  const payload: FhirToMedicalRecordPayload = {
+    fileName,
+    patientId: patient.id,
+    firstName: patient.data.firstName,
+    cxId: patient.cxId,
+    dateFrom,
+    dateTo,
+    conversionType,
+  };
+
+  const lambdaResult = await lambdaClient
+    .invoke({
+      FunctionName: lambdaName,
+      InvocationType: "RequestResponse",
+      Payload: JSON.stringify(payload),
+    })
+    .promise();
+
+  if (lambdaResult.StatusCode !== 200) {
+    throw new MetriportError("Lambda invocation failed", undefined, {
+      lambdaName,
+      status: lambdaResult.StatusCode,
+      log: logResultToString(lambdaResult.LogResult),
+      payload: lambdaResult.Payload?.toString(),
+    });
+  }
+  if (!lambdaResult.Payload) {
+    throw new MetriportError("Payload is undefined", undefined, {
+      lambdaName,
+      status: lambdaResult.StatusCode,
+      log: logResultToString(lambdaResult.LogResult),
+    });
+  }
+
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  const url = lambdaResult.Payload.toString();
+
+  return url.replace(/['"]+/g, "");
+}
+
+// TODO review this logic, ideally we'd have these parameter stored on the S3 object's metadata and not on the filename
+function createFileName({
+  cxId,
+  patientId,
+  resources,
+  dateFrom,
+  dateTo,
+}: {
+  cxId: string;
+  patientId: string;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+}): string {
+  const MEDICAL_RECORD_KEY = "MR";
+  let fileName = `${cxId}/${patientId}/${cxId}_${patientId}_${MEDICAL_RECORD_KEY}`;
+
+  if (resources) {
+    fileName = `${fileName}_${resources.toString()}`;
+  }
+
+  if (dateFrom) {
+    fileName = `${fileName}_${dateFrom}`;
+  }
+
+  if (dateTo) {
+    fileName = `${fileName}_${dateTo}`;
```

Comment:
if dateTo is provided, but dateFrom isn't, how are you going to know which one to identify it as? For these reasons, I added extra information at the end of the s3 key in the form of query params (i.e. `bucket/key/?dateFrom=${dateFrom}&dateTo=${dateTo}`)

Reply:
Yeah, I did the same question on the PR before this was merged... :/

Taken [to Slack](https://metriport.slack.com/archives/C04DMKE9DME/p1697838096984539?thread_ts=1697833766.796829&cid=C04DMKE9DME) as there are more considerations.